core/scripts/vrfv2plus/testnet/main.go

```diff
@@ -411,6 +413,57 @@ func main() {
 			fmt.Println("received receipt, continuing")
 		}
 		fmt.Println("done")
+	case "trusted-bhs-store":
+		cmd := flag.NewFlagSet("trusted-bhs-backwards", flag.ExitOnError)
+		trustedBHSAddr := cmd.String("trusted-bhs-address", "", "address of the trusted bhs contract")
+		blockNumbersString := cmd.String("block-numbers", "", "comma-separated list of block numbers e.g 123,456 ")
+		batchSizePtr := cmd.Int64("batch-size", -1, "batch size")
+		helpers.ParseArgs(cmd, os.Args[2:], "trusted-bhs-address", "end-block", "batch-size", "block-numbers")
+
+		// Parse batch size.
+		batchSize := int(*batchSizePtr)
+
+		// Parse block numbers.
+		blockNumbersStrings := strings.Split(*blockNumbersString, ",")
+		var blockNumbers []*big.Int
+		for _, s := range blockNumbersStrings {
+			b, err := strconv.Atoi(s)
+			helpers.PanicErr(err)
+			blockNumbers = append(blockNumbers, big.NewInt(int64(b)))
+		}
+
+		// Instantiate trusted bhs.
+		trustedBHS, err := trusted_blockhash_store.NewTrustedBlockhashStore(common.HexToAddress(*trustedBHSAddr), e.Ec)
+		helpers.PanicErr(err)
+
+		for i := 0; i < len(blockNumbers); i += batchSize {
+			// Get recent blockhash and block number anew each iteration. We do this so they do not get stale.
+			recentBlockNumber, err := e.Ec.BlockNumber(context.Background())
+			helpers.PanicErr(err)
+			recentBlock, err := e.Ec.BlockByNumber(context.Background(), big.NewInt(int64(recentBlockNumber)))
```

Comment:
Low prio: `HeaderByNumber` is lower bandwidth and probably faster (block by number has to fetch transactions, receipts, etc.)